roxie/ccd/ccd.hpp

```diff
@@ -624,9 +638,26 @@ class ContextLogger : implements IRoxieContextLogger, public CInterface
     virtual void CTXLOGva(const LogMsgCategory & cat, const LogMsgJobInfo & job, LogMsgCode code, const char *format, va_list args) const override  __attribute__((format(printf,5,0))) 
     {
         StringBuffer text, prefix;
-        getLogPrefix(prefix);
         text.valist_appendf(format, args);
+        unsigned now = msTick();
+        CriticalBlock b(crit);
+        if (format == lastFormat && now-lastLogTime < minLogInterval && strsame(text, lastText))
```

Comment:
Does this need to check format (and is it safe)?  If it is an efficiency check there should be a comment - with a comment about requiring constant formats.

Reply:
Should this be a feature of the logging manager instead?